docs/examples/RESTCONF/index.md

```diff
@@ -0,0 +1,550 @@
+
+```shell
+curl -s GET 'https://10.73.1.105:6020/restconf/data/openconfig-interfaces:interfaces/interface=Ethernet4/config' \
+   --header 'Accept: application/yang-data+json' -u arista:arista  --insecure
+```
+
+output
+
+```shell
+{"openconfig-interfaces:description":"blabla","openconfig-interfaces:enabled":false,"arista-intf-augments:load-interval":300,"openconfig-interfaces:loopback-mode":false,"openconfig-interfaces:mtu":0,"openconfig-interfaces:name":"Ethernet4","openconfig-vlan:tpid":"openconfig-vlan-types:TPID_0X8100","openconfig-interfaces:type":"iana-if-type:ethernetCsmacd"}
+```
+
+Let's use the file [interface.json](interface.json)
+
+```shell
+more interface.json
+```
+
+output
+
+```shell
+{"enabled":true,"name":"Ethernet4", "description":"restconf_test"}
+```
+
+```shell
+curl -X PUT https://10.73.1.105:6020/restconf/data/openconfig-interfaces:interfaces/interface=Ethernet4/config \
+   -H 'Content-Type: application/json' -u arista:arista -d @interface.json  --insecure
+```
+
+output
+
+```shell
+{"openconfig-interfaces:description":"restconf_test","openconfig-interfaces:enabled":true,"openconfig-interfaces:name":"Ethernet4"}
+```
+
+Let's verify after the change
+
+```shell
+curl -s GET 'https://10.73.1.105:6020/restconf/data/openconfig-interfaces:interfaces/interface=Ethernet4/config' \
+   --header 'Accept: application/yang-data+json' -u arista:arista  --insecure
+```
+
+output
+
+```shell
+{"openconfig-interfaces:description":"restconf_test","openconfig-interfaces:enabled":true,"arista-intf-augments:load-interval":300,"openconfig-interfaces:loopback-mode":false,"openconfig-interfaces:mtu":0,"openconfig-interfaces:name":"Ethernet4","openconfig-vlan:tpid":"openconfig-vlan-types:TPID_0X8100","openconfig-interfaces:type":"iana-if-type:ethernetCsmacd"}
+```
+
+```shell
+curl -s GET 'https://10.73.1.105:6020/restconf/data/openconfig-interfaces:interfaces/interface=Ethernet4/config' \
+   --header 'Accept: application/yang-data+json' -u arista:arista  --insecure | jq .
+```
+
+output
+
+```shell
+{
+  "openconfig-interfaces:description": "restconf_test",
+  "openconfig-interfaces:enabled": true,
+  "arista-intf-augments:load-interval": 300,
+  "openconfig-interfaces:loopback-mode": false,
+  "openconfig-interfaces:mtu": 0,
+  "openconfig-interfaces:name": "Ethernet4",
+  "openconfig-vlan:tpid": "openconfig-vlan-types:TPID_0X8100",
+  "openconfig-interfaces:type": "iana-if-type:ethernetCsmacd"
+}
+```
+
+#### Device hostname example
+
+Let's check before the change
+
+```shell
+curl -X GET https://10.73.1.105:6020/restconf/data/system/config \
+   --header 'Accept: application/yang-data+json' -u arista:arista  --insecure
+```
+
+output
+
+```shell
+{"openconfig-system:hostname":"DC1-LEAF1A"}
+```
+
+```shell
+curl -X GET https://10.73.1.105:6020/restconf/data/system \
+   --header 'Accept: application/yang-data+json' -u arista:arista \
+   --insecure | jq .'"openconfig-system:config".hostname'
+```
+
+output
+
+```shell
```

Comment:
just a quick thought - i wonder if this might render a bit better as type "text"?

Reply:
I will try

Reply:
done

Reply:
done